test/NuGet.Server.Tests/Infrastructure/TemporaryDirectory.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
+using System;
+using System.IO;
+
+namespace NuGet.Server.Tests.Infrastructure
+{
+    public class TemporaryDirectory 
+        : IDisposable
+    {
+        public string Path { get; private set; }
+
+        public TemporaryDirectory()
+        {
+            var tempPath = System.IO.Path.GetTempPath();
```

Comment:
Don't put stuff directly in the temp folder. Create a subfolder like the nuget.client tests do.